lib/ace/http1/endpoint.ex

```diff
@@ -56,9 +58,18 @@ defmodule Ace.HTTP1.Endpoint do
           send(state.worker, {state.channel, part})
         end)
 
-        :ok = Ace.Socket.set_active(state.socket)
-        timeout = if HTTP1.Parser.done?(receive_state), do: :infinity, else: @packet_timeout
-        {:noreply, %{state | receive_state: receive_state}, timeout}
+        pending_ack_count = state.pending_ack_count + Enum.count(parts)
+
+        timeout =
+          if pending_ack_count < @max_pending_request_part_count do
+            :ok = Ace.Socket.set_active(state.socket)
+            if HTTP1.Parser.done?(receive_state), do: :infinity, else: @packet_timeout
+          else
+            :infinity
```

Comment:
:thinking: 

I guess it's fair to return infinity here. the packet timeout is to protect against slow attacks but anyone sending data faster than it is processed can hardly count as a slow attack